src/mon/OSDMonitor.cc

```diff
@@ -10193,6 +10195,10 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
       }
     } else {
       //NOTE:for replicated pool,cmd_map will put rule_name to erasure_code_profile field
+      //     and put expected_num_objects to ruleset field
+      if (rule_name != "") {
+        expected_num_objects = std::atoll(rule_name.c_str());
```

Comment:
you should be able to get the value with `cmd_getval()`.

Reply:
As 'rule' is stored as string, we can not get its value as a int64_t type with cmd_getval.
And rule already stored the string, so we only need to convert it to int64_t.

Reply:
good point. Please use `strict_strtoll()` as is used in other parts of `OSDMonitor.cc`.

Reply:
ok